File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ConnectionErrorHandler.java
package closer.vlllage.com.closer.handler.helpers;

import closer.vlllage.com.closer.pool.PoolMember;

public class ConnectionErrorHandler extends PoolMember {

    private OnConnectionErrorListener onConnectionErrorListener;

    public void setOnConnectionErrorListener(OnConnectionErrorListener onConnectionErrorListener) {
        this.onConnectionErrorListener = onConnectionErrorListener;
    }

    public void connectionError() {
        if (onConnectionErrorListener != null) {
            onConnectionErrorListener.onConnectionError();
            return;
        }

        $(DefaultAlerts.class).syncError();
    }

    public interface OnConnectionErrorListener {
        void onConnectionError();
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ConnectionErrorHandler.kt
package closer.vlllage.com.closer.handler.helpers

import com.queatz.on.On

class ConnectionErrorHandler constructor(private val on: On) {

    var onConnectionErrorListener: (() -> Unit)? = null

    fun notifyConnectionError() {
        if (onConnectionErrorListener != null) {
            onConnectionErrorListener!!.invoke()
            return
        }

        on<DefaultAlerts>().syncError()
    }
}
